### PR TITLE
Display setup message if wifi manager is in setup mode

### PR DESCRIPTION
**Pull request by murmeltier08**
Display setup message if wifi manager is in setup mode

### CODE REVIEW

File: src/Storage.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ static const sto_data_t itemDefaults PROGMEM = {
     BREW_SW_TIME,                             // STO_ITEM_BREW_SW_TIME
     {0xFF, 0xFF},                             // free to use
     BREWSENSITIVITY,                          // STO_ITEM_BD_THRESHOLD
-    0xFF,                                     // free to use
+    WIFISETUP,                                     // WIFI SETUP
```
Comment (kjyv) [inline src/Storage.cpp:95]:
```suggestion
    WIFISETUP,                                // WIFI SETUP
```

File: src/Storage.cpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ typedef struct __attribute__((packed)) {
     double brewSwTimeSec;
     uint8_t freeToUse10[2];
     double brewDetectionThreshold;
-    uint8_t freeToUse11;
+    double wifisetup;
```
Comment (kjyv) [inline src/Storage.cpp:49]:
```suggestion
    double wifiCredentialsSaved;
```

File: src/Storage.cpp (hunk 1 above)
Comment (kjyv) [reply src/Storage.cpp:95]: and change to WIFI_CREDENTIALS_SAVED etc. in all places

File: src/Storage.cpp (hunk 2 above)
Comment (kjyv) [reply src/Storage.cpp:49]: I think "wifisetup" is a bit too ambiguous

Comment (kjyv) [reply src/Storage.cpp:49]: Oh and I guess double is not a good idea, use uint8_t

File: src/main.cpp [hunk 3]
```diff
@@ -1345,13 +1347,27 @@ void wiFiSetup() {
...
+            
+            const char hostname[] = (STR(HOSTNAME)) ;
+            displayLogo("connect to  AP: ",HOSTNAME);
+            debugPrintf("connect to  AP: %s \n",String(hostname));
+    }
     wm.setHostname(hostname);
 
     if (wm.autoConnect(hostname, pass)) {
+        WifiSetup = 1;
+        Serial.println(WifiSetup);
```
Comment (kjyv) [inline src/main.cpp:1366]: if this debug output should stay, use debugPrintln instead for OTA logging

File: src/main.cpp [hunk 4]
```diff
@@ -1345,13 +1347,27 @@ void wiFiSetup() {
...
+    //wm.setWiFiAutoReconnet(true);
+    // Wifisetup fro
+    sysParaWifiSetup.getStorage() ;
+    Serial.println("XXXXXXXX");
+    Serial.println(WifiSetup);
+    if(WifiSetup == 0)
+    {
+            
+            const char hostname[] = (STR(HOSTNAME)) ;
+            displayLogo("connect to  AP: ",HOSTNAME);
```
Comment (kjyv) [inline src/main.cpp:1359]:
```suggestion
            displayLogo("Connect to Wifi: ", HOSTNAME);
```

Comment (kjyv) [reply src/main.cpp:1359]: should also use translation strings though

File: src/main.cpp [hunk 5]
```diff
@@ -1345,13 +1347,27 @@ void wiFiSetup() {
     wm.setConnectTimeout(10); // Try 10 sec to connect to WLAN, 5 sec too short!
     wm.setBreakAfterConfig(true);
     wm.setConnectRetries(3);
-    //wm.setWiFiAutoReconnect(true);
+    //wm.setWiFiAutoReconnet(true);
```
Comment (kjyv) [inline src/main.cpp:1350]:
```suggestion
    //wm.setWiFiAutoReconnect(true);
```

File: src/Storage.cpp (hunk 2 above)
Comment (murmeltier08) [reply src/Storage.cpp:49]: good idea with the name!

Comment (murmeltier08) [reply src/Storage.cpp]: Implemented in the last commit

File: src/main.cpp (hunk 4 above)
Comment (murmeltier08) [reply src/main.cpp:1357]: Implemented in the last commit

File: src/main.cpp (hunk 5 above)
Comment (murmeltier08) [reply src/main.cpp]: Implemented in the last commit

**Comment by murmeltier08** [comment]
I consider the suggestions from the review in the last commit

File: src/main.cpp [hunk 6]
```diff
@@ -1348,12 +1350,23 @@ void wiFiSetup() {
     wm.setBreakAfterConfig(true);
     wm.setConnectRetries(3);
     //wm.setWiFiAutoReconnect(true);
+    // Wifisetup fro
+    sysParaWifiCredentialsSaved.getStorage() ;
```
Comment (fiendie) [inline src/main.cpp:1354]: Unnecessary space before semicolon

**Comment by murmeltier08** [comment]
I fix a MQTT failure of wrong variable, Display shows now complete display version. Wifi setup is shown in Display.

File: src/main.cpp [hunk 7]
```diff
@@ -1348,12 +1350,23 @@ void wiFiSetup() {
...
+    if(WifiCredentialsSaved == 0)
+    {
+            
+            const char hostname[] = (STR(HOSTNAME)) ;
```
Comment (fiendie) [inline src/main.cpp]: See above.

File: src/Storage.cpp [hunk 8]
```diff
@@ -47,7 +47,7 @@ typedef struct __attribute__((packed)) {
     double brewPIDDelaySec; 
     uint8_t freeToUse10;
     double brewDetectionThreshold;
-    double wifisetup;
+    uint8_t WifiCredentialsSaved;
```
Comment (kjyv) [inline src/Storage.cpp:50]: good but needs to start lowercase, capital letters are usually reserved for class names ;)

File: src/defaults.h [hunk 9]
```diff
@@ -40,7 +40,7 @@ int writeSysParamsToStorage(void);
 #define PRE_INFUSION_TIME 2        // pre-infusion time in seconds
 #define PRE_INFUSION_PAUSE_TIME 5  // pre-infusion pause time in seconds
 #define SCALE_WEIGHTSETPOINT 30    // Target weight in grams
-#define WIFISETUP 0                 // Flag if Wifisetup is done 0: no setup, 1: setup via wifi manager
+#define WIFI_CREDENTIALS_SAVED 0   // Flag if Wifisetup is done 0: no setup, 1: setup via wifi manager
```
Comment (kjyv) [inline src/defaults.h:43]:
```suggestion
#define WIFI_CREDENTIALS_SAVED 0   // Flag if wifi setup is done. 0: not setup, 1: credentials set up via wifi manager
```

File: src/main.cpp [hunk 10]
```diff
@@ -1349,24 +1349,20 @@ void wiFiSetup() {
...
-    Serial.println(WifiSetup);
-    if(WifiSetup == 0)
-    {
-            
-            const char hostname[] = (STR(HOSTNAME)) ;
-            displayLogo("connect to  AP: ",HOSTNAME);
-            debugPrintf("connect to  AP: %s \n",String(hostname));
+    sysParaWifiCredentialsSaved.getStorage();
+
+    if(WifiCredentialsSaved == 0){
```
Comment (kjyv) [inline src/main.cpp:1356]:
```suggestion
    if (WifiCredentialsSaved == 0) {    
```

File: src/main.cpp [hunk 11]
```diff
@@ -1349,24 +1349,20 @@ void wiFiSetup() {
...
-    {
-            
-            const char hostname[] = (STR(HOSTNAME)) ;
-            displayLogo("connect to  AP: ",HOSTNAME);
-            debugPrintf("connect to  AP: %s \n",String(hostname));
+    sysParaWifiCredentialsSaved.getStorage();
+
+    if(WifiCredentialsSaved == 0){    
+        const char hostname[] = (STR(HOSTNAME));
+        displayLogo("Connect to Wifi: ",HOSTNAME);
```
Comment (kjyv) [inline src/main.cpp:1358]:
```suggestion
        displayLogo("Connect to Wifi: ", HOSTNAME);
```

File: src/main.cpp [hunk 12]
```diff
@@ -1349,24 +1349,20 @@ void wiFiSetup() {
...
+        debugPrintf("Connect to Wifi: %s \n",String(hostname));
```
Comment (kjyv) [inline src/main.cpp:1359]:
```suggestion
        debugPrintf("Connect to Wifi: %s \n", String(hostname));
```

File: src/main.cpp [hunk 13]
```diff
@@ -2274,10 +2270,9 @@ int readSysParamsFromStorage(void) {
     if (sysParaPreInfTime.getStorage() != 0) return -1;
     if (sysParaPreInfPause.getStorage() != 0) return -1;
     if (sysParaPidKpSteam.getStorage() != 0) return -1;
-    if (sysParaSteamSetPoint.getStorage() != 0) return -1;
-    if (sysParaUsePonM.getStorage() != 0) return -1;
```
Comment (kjyv) [inline src/main.cpp]: why remove these lines?

File: src/main.cpp [hunk 14]
```diff
@@ -2309,12 +2304,10 @@ int writeSysParamsToStorage(void) {
     if (sysParaPreInfTime.setStorage() != 0) return -1;
     if (sysParaPreInfPause.setStorage() != 0) return -1;
     if (sysParaPidKpSteam.setStorage() != 0) return -1;
-    if (sysParaSteamSetPoint.setStorage() != 0) return -1;
-    if (sysParaUseBDPID.setStorage() != 0) return -1;
```
Comment (kjyv) [inline src/main.cpp]: also here?

File: src/Storage.cpp [hunk 15]
```diff
@@ -91,9 +91,10 @@ static const sto_data_t itemDefaults PROGMEM = {
     AGGBTV,                                   // STO_ITEM_PID_TV_BD
     {0xFF, 0xFF},                             // free to use
     BREW_SW_TIME,                             // STO_ITEM_BREW_SW_TIME
-    {0xFF, 0xFF},                             // free to use
-    BREWSENSITIVITY,                          // STO_ITEM_BD_THRESHOLD
-    WIFISETUP,                                     // WIFI SETUP
+    BREW_PID_DELAY,                           // STO_ITEM_BREW_PID_DELAY
```
Comment (kjyv) [inline src/Storage.cpp:94]: how do these changes show up here?

File: src/Storage.h [hunk 16]
```diff
@@ -31,9 +31,11 @@ typedef enum
   STO_ITEM_BREW_SW_TIME,            // brew software time
   STO_ITEM_BREW_PID_DELAY,          // brew PID delay
   STO_ITEM_BD_THRESHOLD,            // brew detection limit
-  STO_ITEM_WIFISETUP,               // flag for wifisetup
-  STO_ITEM_PID_KP_START,            // PID P part at cold start phase
-  STO_ITEM_PID_TN_START,            // PID I part at cold start phase
+  STO_ITEM_WIFI_CREDENTIALS_SAVED,  // flag for wifisetup
+  STO_ITEM_PRE_INFUSION_TIME,       // pre-infusion time
+  STO_ITEM_PRE_INFUSION_PAUSE,      // pre-infusion pause
```
Comment (kjyv) [inline src/Storage.h:36]: and where do these come from, these lines should have been in the code already?

File: src/main.cpp [hunk 17]
```diff
@@ -218,6 +219,7 @@ SysPara<uint8_t> sysParaUseBDPID(&useBDPID, 0, 1, STO_ITEM_USE_BD_PID);
 SysPara<double> sysParaBrewTime(&brewtime, BREW_TIME_MIN, BREW_TIME_MAX, STO_ITEM_BREW_TIME);
 SysPara<double> sysParaBrewSwTime(&brewtimesoftware, BREW_SW_TIME_MIN, BREW_SW_TIME_MAX, STO_ITEM_BREW_SW_TIME);
 SysPara<double> sysParaBrewThresh(&brewSensitivity, BD_THRESHOLD_MIN, BD_THRESHOLD_MAX, STO_ITEM_BD_THRESHOLD);
+SysPara<uint8_t> sysParawifiCredentialsSaved(&wifiCredentialsSaved, WIFI_CREDENTIALS_SAVED_MIN, WIFI_CREDENTIALS_SAVED_MAX, STO_ITEM_WIFI_CREDENTIALS_SAVED);
```
Comment (kjyv) [inline src/main.cpp:222]: Ah, here it should stay capitalized of course ;) 
sysParaWifiCredentialsSaved

File: src/main.cpp [hunk 18]
```diff
@@ -1348,12 +1350,22 @@ void wiFiSetup() {
     wm.setBreakAfterConfig(true);
     wm.setConnectRetries(3);
     //wm.setWiFiAutoReconnect(true);
+    // Wifisetup fro
+    sysParaWifiCredentialsSaved.getStorage();
+
+    if(wifiCredentialsSaved == 0){
```
Comment (kjyv) [inline src/main.cpp:1356]:
```suggestion
    if (wifiCredentialsSaved == 0) {
```

File: src/main.cpp [hunk 19]
```diff
@@ -1869,15 +1881,16 @@ void setup() {
         u8g2.setI2CAddress(oled_i2c * 2);
         u8g2.begin();
         u8g2_prepare();
-        displayLogo(String("Version ") + String(sysVersion), "");
-       // delay(2000); // caused crash with wifi manager
+        displayLogo(String("Version "), String(sysVersion));
+        delay(2000); // caused crash with wifi manager
```
Comment (kjyv) [inline src/main.cpp:1885]:
```suggestion
        delay(2000); // caused crash with wifi manager on esp8266, should be ok on esp32
```

File: src/main.cpp [hunk 20]
```diff
@@ -1869,15 +1881,16 @@ void setup() {
...
     #endif
 
     // Init Scale by BREWMODE 2 or SHOTTIMER 2
     #if (BREWMODE == 2 || ONLYPIDSCALE == 1)
         initScale();
     #endif
 
+
```
Comment (kjyv) [inline src/main.cpp:1893]:
```suggestion
```

**Pull request merged**